modules/vaos/app/serializers/vaos/appointment_requests_serializer.rb

```diff
@@ -4,14 +4,23 @@ module VAOS
   class AppointmentRequestsSerializer
     include FastJsonapi::ObjectSerializer
 
-    set_id do |object|
-      object.data_identifier[:unique_id]
-    end
+    set_id(&:appointment_request_id)
 
     set_type :appointment_requests
 
     attribute :facility do |object|
-      object.facility.reverse_merge(type: nil, address: nil)
+      object.facility.reverse_merge(type: nil, address: nil).except(:link, :object_type)
+    end
+
+    attribute :patient do |object|
+      object.patient.slice(:inpatient, :text_messaging_allowed)
+    end
+
+    attribute :appointment_request_detail_code do |object|
+      Array.wrap(object.appointment_request_detail_code).map do |element|
+        element.slice(:appointment_request_detail_code_id, :created_date)
+               .merge(detail_code: element[:detail_code].slice(:code, :provider_message, :veteran_message))
+      end
     end
```

Comment:
What in the world is happening here?? 😎

Reply:
Is there some way to make this more readable?

Reply:
yep, this was merged in from another branch should be cleaner now

Reply:
basically appointment_request_detail_code is an array of complex object that includes a bunch of stuff including patient identifiers that we'd like to NOT pass up to the FE. 

Perhaps it could have its own serializer in the future, but we'll revisit that when we refactor.

Reply:
i like it 👍 
although the double slice caught me off-guard.
seems pretty rubyish

Reply:
it's technically more performant than doing a merge! or slice! when iterating a large array. I'm not sure that the changes made make it any more "readable." The best way to address readability would be to give its own serializer and we'll get to that eventually when we refactor. This would have the added benefit of ensuring that the data coming in doesn't include some key we've never seen before that includes some kind of PII we don't want to expose to the FE. We're doing a ton of regression testing and haven't seen anything like that yet (and apparently it's ok in the legacy app we're replacing even if we were to find it), but something we'd prefer to avoid where possible.